### PR TITLE
Check this puppy out 

### PR DESCRIPTION
**Pull request by litehed**

### CODE REVIEW

**Comment by litehed** [comment]
Wait shoot I messed some stuff up I redact that Check this puppy out statement

**Comment by JIceberg** [comment]
Too late

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -5,9 +5,9 @@ name: Java CI with Gradle
 
 on:
   push:
-    branches: [ main ]
+    branches: '**'
   pull_request:
-    branches: [ main ]
+    branches: '**'
```
Comment (JIceberg) [inline .github/workflows/gradle.yml:10]: PagChomp

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOpMain.java [hunk 2]
```diff
@@ -45,7 +57,9 @@ public void initialize() {
         bR = new Motor(hardwareMap, "bR");
 
         shot = new MotorEx(hardwareMap, "shot", Motor.GoBILDA.BARE);
+        pickup = new MotorEx(hardwareMap, "wobble", Motor.GoBILDA.BARE);
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOpMain.java:65]: Why exactly is the pickup motor bare?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/IntakeSystem.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import com.arcrobotics.ftclib.command.SubsystemBase;
+import com.arcrobotics.ftclib.hardware.motors.Motor;
+
+import org.firstinspires.ftc.robotcore.external.Telemetry;
+
+
+public class IntakeSystem extends SubsystemBase {
+    private Motor intakeMotor;
+    private Telemetry telemetry;
+    public boolean intakeActive = false;
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/IntakeSystem.java]: Make this private.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/IntakeSystem.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    public IntakeSystem(Motor IntakeMotor) {
+        intakeMotor = IntakeMotor;
+    }
+
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/IntakeSystem.java:21]: Add two methods. One called `toggle()` and one called `active()`:
```java
public boolean active() {
    return intakeActive;
}

public void toggle() {
    intakeActive = !intakeActive;
}
```

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOpMain.java [hunk 5]
```diff
@@ -75,15 +89,26 @@ public void initialize() {
...
+        pickupCommand = new Com_PickUp(wobbleSystem);
+        goalLift = (new GamepadButton(m_driverOp, GamepadKeys.Button.Y))
+                .whenPressed(pickupCommand);
+
+        intakeSystem = new IntakeSystem(intake);
+        startIntakeCommand = new Com_IntakeStart(intakeSystem);
+        stopIntakeCommand = new Com_IntakeStop(intakeSystem);
+        toggleIntake = intakeSystem.intakeActive ? new GamepadButton(m_driverOp, GamepadKeys.Button.X)
+                .whenPressed(stopIntakeCommand) : new GamepadButton(m_driverOp, GamepadKeys.Button.X)
+                .whenPressed(startIntakeCommand);
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/TeleOpMain.java]: I don't think this works. I'd recommend using one singular definition for the button with a ConditionalCommand. and multiple bindings. My suggestion is this:
```java
toggleIntake = new GamepadButton(m_driverOp, GamepadKeys.Button.X)
        .whenPressed(new ConditionalCommand(startIntakeCommand, stopIntakeCommand, intakeSystem::active))
        .whenReleased(new InstantCommand(intakeSystem::toggle));
```
Refer to my comment in IntakeSubsystem for more information.

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/WobbleSystem.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+
+public class WobbleSystem extends SubsystemBase {
+    MotorEx rundhi;
+    public WobbleSystem(MotorEx pickMeUpDaddy){
+        rundhi = pickMeUpDaddy;
+    }
+
+    public void spinMeRightRoundBaby(){
+        //I need to test stuff out so Im leaving this unfinished for now
```
Comment (JIceberg) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/WobbleSystem.java]: Make this `//TODO: testing`

**Review by JIceberg — CHANGES_REQUESTED** [review]

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/IntakeSystem.java (hunk 3 above)
Comment (JIceberg) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/IntakeSystem.java]: Also, instantiate it as true.

**Comment by litehed** [comment]
Lol neither of us noticed that I never declared the motor for the intake

**Comment by JIceberg** [comment]
You never answered when I asked why the wobble goal motor is bare.

**Comment by litehed** [comment]
> You never answered when I asked why the wobble goal motor is bare.

Oh yeah I don't know what it's supposed to be and I was using a 6k for testing which is also why I don't have code in the spin me right round baby

**Comment by JIceberg** [comment]
Ah okay. We'll leave it then. Ask Aiden when you get the chance.

**Review by JIceberg — APPROVED** [review]
Nice.

**Pull request merged**